apps/ui/src/composables/useSharing.ts

```diff
@@ -0,0 +1,82 @@
+import { FunctionalComponent } from 'vue';
+import { Proposal } from '@/types';
+import ICFarcaster from '~icons/c/farcaster';
+import ICLens from '~icons/c/lens';
+import ICX from '~icons/c/x';
+
+export type ShareableType = 'proposal' | 'user';
+type SocialNetwork = 'x' | 'lens' | 'farcaster';
+
+const HASH_TAG = 'Snapshot';
+
+const SOCIAL_NETWORKS: {
+  id: SocialNetwork;
+  name: string;
+  icon: FunctionalComponent;
+}[] = [
+  { id: 'x', name: 'X', icon: ICX },
+  { id: 'lens', name: 'Hey', icon: ICLens },
+  { id: 'farcaster', name: 'Warpcast', icon: ICFarcaster }
+];
+
+export function useSharing() {
+  function getProposalUrl(proposal: Proposal) {
+    return `https://${window.location.hostname}/#/${proposal.network}:${proposal.space.id}/proposal/${proposal.proposal_id}`;
+  }
+
+  function getMessage(type: ShareableType, payload: any) {
+    switch (type) {
+      case 'user':
+        return getUserMessage();
+      case 'proposal':
+        return getProposalMessage(payload as Proposal);
+      default:
+        throw new Error('Invalid shareable object');
+    }
+  }
+
+  function getUserMessage() {
+    return encodeURIComponent(window.location.href);
+  }
+
+  function getProposalMessage(proposal: Proposal) {
+    return encodeURIComponent(
+      `${proposal.space.name}: ${proposal.title} ${getProposalUrl(proposal)}`
+    );
+  }
+
+  function getShareUrl(
```

Comment:
maybe `getSharableURL`

Reply:
I prefer the short `getShareUrl` better. shareable mean either `Proposal` or `User`, etc ....

At this point, we only care about the url, and not if it's a proposal or user url

Reply:
> shareable mean either Proposal or User, etc ...

hmm we can go with `getShareUrl` , but we do have `type: ShareableType` below this 😅 which is either `Proposal` or `User`